### PR TITLE
[CLEANUP] Stop using tmpl->getFileName

### LINKED ISSUE #470: Stop using tmpl->getFileName
Either in general, or at least in TYPO3 V9.

### PR DESCRIPTION
**Pull request by Konafets**
Closes #470

### CODE REVIEW

File: Classes/Configuration/ConfigurationCheck.php [hunk 1]
```diff
@@ -56,6 +57,11 @@ class ConfigurationCheck
      */
     private $errorText = '';
 
+    /**
+     * @var FilePathSanitizer $filePathSanitizer
```
Comment (oliverklee) [inline Classes/Configuration/ConfigurationCheck.php:61]: The variable name should not be here in the annotation. It's only required for local variables, not for properties.

File: Classes/Configuration/ConfigurationCheck.php [hunk 2]
```diff
@@ -373,7 +383,7 @@ protected function checkCssFileFromConstants()
         $typoScriptSetupPage = &$frontEndController->tmpl->setup['page.'];
         $fileName = $typoScriptSetupPage['includeCSS.'][$this->objectToCheck->prefixId];
         if (!empty($fileName)) {
-            $fileName = $frontEndController->tmpl->getFileName($fileName);
+            $fileName = $this->filePathSanitizer->sanitize($fileName);
```
Comment (oliverklee) [inline Classes/Configuration/ConfigurationCheck.php:386]: The local variable `$frontEndController` then is not needed anymore.

File: Classes/Configuration/ConfigurationCheck.php [hunk 3]
```diff
@@ -331,7 +338,10 @@ protected function checkTemplateFile(bool $canUseFlexforms = false)
                 's_template_special',
                 true
             );
-            if (!is_file($this->getFrontEndController()->tmpl->getFileName($rawFileName))) {
+
+            $file = $this->filePathSanitizer->sanitize($rawFileName);
+
```
Comment (oliverklee) [inline Classes/Configuration/ConfigurationCheck.php:343]: Please let*s drop this blank line here:

File: Classes/Configuration/ConfigurationCheck.php [hunk 4]
```diff
@@ -331,7 +338,10 @@ protected function checkTemplateFile(bool $canUseFlexforms = false)
                 's_template_special',
                 true
             );
-            if (!is_file($this->getFrontEndController()->tmpl->getFileName($rawFileName))) {
+
+            $file = $this->filePathSanitizer->sanitize($rawFileName);
```
Comment (oliverklee) [inline Classes/Configuration/ConfigurationCheck.php:342]: We then also do not need the null check for the FE controller a few lines up.

**Comment by Konafets** [comment]
Will fix this one later.

File: Classes/Templating/TemplateHelper.php [hunk 5]
```diff
@@ -513,7 +514,7 @@ public function getTemplateCode(bool $ignoreFlexform = false)
         );
 
         if (!$ignoreFlexform) {
-            $templateFileName = (string)$this->getFrontEndController()->tmpl->getFileName($templateFileName);
+            $templateFileName = GeneralUtility::makeInstance(FilePathSanitizer::class)->sanitize($templateFileName);
```
Comment (oliverklee) [inline Classes/Templating/TemplateHelper.php:517]: Please let's put the sanitizer in  a local variable so it can have an `@var` annotation.

**Review by oliverklee — CHANGES_REQUESTED** [review]

**Review by oliverklee — CHANGES_REQUESTED** [review]
These changes seem to need version checks for 8/9.

File: Classes/Configuration/ConfigurationCheck.php [hunk 6]
```diff
@@ -331,7 +332,15 @@ protected function checkTemplateFile(bool $canUseFlexforms = false)
                 's_template_special',
                 true
             );
-            if (!is_file($this->getFrontEndController()->tmpl->getFileName($rawFileName))) {
+
+            if (Typo3Version::isNotHigherThan(8)) {
+                $file = $this->getFrontEndController()->tmpl->getFileName($rawFileName);
+            } else {
+                $file = GeneralUtility::makeInstance(\TYPO3\CMS\Frontend\Resource\FilePathSanitizer::class)
```
Comment (oliverklee) [inline Classes/Configuration/ConfigurationCheck.php:339]: Please `use` the class.

File: Classes/Templating/TemplateHelper.php [hunk 7]
```diff
@@ -513,7 +513,12 @@ public function getTemplateCode(bool $ignoreFlexform = false)
         );
 
         if (!$ignoreFlexform) {
-            $templateFileName = (string)$this->getFrontEndController()->tmpl->getFileName($templateFileName);
+            if (Typo3Version::isNotHigherThan(8)) {
+                $templateFileName = (string)$this->getFrontEndController()->tmpl->getFileName($templateFileName);
+            } else {
+                $templateFileName = GeneralUtility::makeInstance(\TYPO3\CMS\Frontend\Resource\FilePathSanitizer::class)
```
Comment (oliverklee) [inline Classes/Templating/TemplateHelper.php:519]: Please `use` the class.

**Review by oliverklee — CHANGES_REQUESTED** [review]
Some of the older comments also are not addressed yet.

**Review by oliverklee — APPROVED** [review]

**Pull request merged**